src/cert_chain.rs

```diff
@@ -34,14 +35,27 @@ impl CertChainContext {
     /// https://msdn.microsoft.com/de-de/library/windows/desktop/aa377182(v=vs.85).aspx
     /// rgpChain[cChain - 1] is the final chain
     pub fn final_chain(&self) -> Option<CertChain> {
-        let cloned = self.clone();
-        let chains = unsafe {
-            let cert_chain = *cloned.0;
+        if let Some(chain) = self.chains().last(){
+            return Some(CertChain(chain.0, self.clone()));
+        }
+        None
+    }
+    
+    pub fn get_chain(&self, index :usize) -> Option<CertChain> {
```

Comment:
We'll want some way to know how many chains there are if we have this, and the implementation shouldn't be a linear scan.